### PR TITLE
M3 linreg

### PR DESCRIPTION
**Pull request by cmmclaug**
Heads up that this is draft PR to give a template on some of the stuff I've been working on.

Outstanding questions:
* Do we want to go full mimic of the sklearn interface and deviate from Python?
* As is there are no input assertions yet
* No tests have been written

### CODE REVIEW

**Review by danielon-5 — APPROVED** [review]
looks good Craig

File: R/aridanalysis.R [hunk 1]
```diff
@@ -22,14 +22,100 @@ arid_eda <- function(data_frame, response, response_type, features=c())
...
+    
+  thisEnv <- environment()
+  assign("regularization_", regularization, thisEnv)
+  assign("lambda_", lambda, thisEnv)
+
+  y <- df %>%
+    dplyr::select({{response}}) %>%
+    as.matrix()
+    
+  cat_features <- tdf %>%
```
Comment (nphaterp) [inline R/aridanalysis.R]:
```suggestion
  cat_features <- df %>% 
```

File: R/aridanalysis.R [hunk 2]
```diff
@@ -22,14 +22,100 @@ arid_eda <- function(data_frame, response, response_type, features=c())
...
+    dplyr::select(-y) %>%
+    dplyr::select_if(is.character) %>%
+    data.matrix()
+
+  numeric_features <- tdf %>%
```
Comment (nphaterp) [inline R/aridanalysis.R]:
```suggestion
  numeric_features <- df %>%
```

**Review by ansarusc — CHANGES_REQUESTED** [review]
So far so good Craig, some important points:
1. Please add a more step-by-step example on how to use the class instance created.
2. Make sure to combine the initial `assign` methods in a function.
3. In order to mimic sklearn correctly, the function should receive X and y, it would be ideal to remove the initial wrangling and let the used provide his/her clean and tidy X and y

**Review by nphaterp — COMMENTED** [review]
nice.

**Pull request merged**